File: Habitica/src/main/java/com/habitrpg/android/habitica/ui/views/insufficientCurrency/InsufficientGoldDialog.java
package com.habitrpg.android.habitica.ui.views.insufficientCurrency;

import android.content.Context;

import com.habitrpg.android.habitica.R;
import com.habitrpg.android.habitica.ui.views.HabiticaIconsHelper;

public class InsufficientGoldDialog extends InsufficientCurrencyDialog {
    public InsufficientGoldDialog(Context context) {
        super(context);

        imageView.setImageBitmap(HabiticaIconsHelper.imageOfGoldReward());
        textView.setText(context.getString(R.string.insufficientGold));
        setTitle(R.string.insufficientSubscriberGemsTitle);
    }
}

File: Habitica/src/main/java/com/habitrpg/android/habitica/ui/views/insufficientCurrency/InsufficientGoldDialog.kt
package com.habitrpg.android.habitica.ui.views.insufficientCurrency

import android.content.Context
import android.os.Bundle
import com.habitrpg.android.habitica.R

class InsufficientGoldDialog(context: Context) : InsufficientCurrencyDialog(context) {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        imageView.setImageResource(R.drawable.gold_multiple)
        textView.text = context.getString(R.string.insufficientGold)
        setTitle(R.string.insufficientGoldTitle)

        addButton(R.string.take_me_back, true)
    }
}
